### ISSUE TITLE
RabbitMQ: nodes are pegged to a single CPU core in production configuration

### ISSUE THREAD

**Issue by michaelklishin** [bitnami/charts#2189]
**Which chart**:

RabbitMQ

**Describe the bug**

These [default production values](https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/values-production.yaml#L100-L104) for [runtime schedulers](https://www.rabbitmq.com/runtime.html#scheduling) are very questionable and peg the nodes
deployed using this chart to a single core, which has obvious implications.

It would make a lot more sense to use as many schedulers as a pod is allowed to use.
If that's not possible to compute, there should be very clear and visible documentation that says
that these values should be modified in actual production deployments.

**Additional context**

 * [RabbitMQ Runtime Tuning guide](https://www.rabbitmq.com/runtime.html#scheduling)

**Comment by rafariossaa** [comment]
Hi @michaelklishin ,

Thanks for noticing this.
In an kubernetes scenario the pod doesn't know about the resources available on the host, so is it not possible to estimate that unless you have full control of the cluster.
I am not a rabbitmq expert, but could it possible that playing with the number of replicas and the affinity will make the trick, or unless, make it better ?

In any case, if you'd like to make a PR, I will be glad of reviewing it .

**Comment by michaelklishin** [comment]
@rafariossaa the number of replicas or affinity won't change the fact that the node is pegged to a single core, which is not how data services are typically run. At the very least there should be a huge warning about this default in the docs: this value is anything but production-optimal.

**Comment by michaelklishin** [comment]
@gerhard @mkuratczyk any suggestions as to how this can be computed or otherwise made more optimal for production workloads? Or is this entirely up to the operator to configure the compute resources first and then this setting?

**Comment by gerhard** [comment]
My recommendation would be to not set the number of schedulers - let Erlang VM discover how many CPUs the system is running. With this approach, for production workloads, I would recommend configuring the K8S scheduling so that a single RabbitMQ pod will end up running per K8S node, and no other pod will be scheduled on that node.

The ideal world for RabbitMQ is pod-vm with NSX networking, which is what @cdickmann & @st3v are currently experimenting with. My expectation is for all K8S flavours (GKE, EKS, AKS, etc.) to eventually behave the same, but until they do, we must avoid Erlang pods with CPU slices and bridged networking (vs host or hardware) for production workloads.

The Erlang VM, like the JVM, expects to have all system CPUs to itself, and it makes many internal optimisations with this assumption. By default, the Erlang VM will start 1 scheduler per CPU. Constraining it to a single CPU, regardless how many CPUs the pod is allowed to utilise, is worse than allowing the default behaviour. `+S 1` is the equivalent of 🤦🏻‍♂️while removing it is the equivalent of 😑, meaning that users will be able to discover via [microstate accounting metrics](https://github.com/deadtrickster/prometheus.erl/pull/98) why their Erlang-based workload performance is terrible.

In conclusion, the simplest thing would be to remove `+S 1` and add documentation on how to configure single RabbitMQ pod per node, with no other pod being scheduled on that node (except system pods).

**Comment by rafariossaa** [comment]
Hi,
  Thank your very much @gerhard  for sharing you knowledge. 

  @michaelklishin ,I will change this to be the default behavior, meanwhile you can comment `rabbitmq.maxAvailableSchedulers` and it will make to not generate the `+S` parameter, as you can see in these [lines]( https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/templates/statefulset.yaml#L236).

**Comment by rafariossaa** [comment]
Hi, 
  I did a small change to fix this issue.
 For the affinity, you will need to uncomment the corresponding block of lines in the `values.yml` or `values-production.yml` and use the appropriate labels for your deployment.

**Comment by michaelklishin** [comment]
Highly relevant here: Erlang/OTP 23 will [take CPU quotas into account](http://blog.erlang.org/OTP-23-Highlights/) when computing default number of runtime schedulers. So the change in #2252 is relevant today but will become a total no-brainer as soon as RabbitMQ Docker images transition to OTP 23 (and it should not take long).

**Comment by rafariossaa** [comment]
Hi @michaelklishin ,
  Thank you very much for letting us know this. 

Best regards,

**Comment by cheskayang** [comment]
@rafariossaa thx a lot for the change. I noticed in the README, the default value for `maxAvailableSchedulers ` and `onlineSchedulers ` are still set as 2 and 1, which could be a bit confusing https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/README.md#rabbitmq-parameters

**Comment by rafariossaa** [comment]
Hi @cheskayang ,
  Not very sure about what is confusing. I don't mean that they are not confusing just I would like to understand how I can fix this. 
  Do you have any suggestion ?.
